pynest/nest/lib/hl_api_connection_helpers.py

```diff
@@ -236,3 +248,61 @@ def fixdict(d):
     projections = fixdict(projections)
     sps(projections)
     sr('ConnectLayers')
+
+
+def _check_input_nodes(pre, post, conn_spec):
+    '''
+    Check the properties of `pre` and `post` nodes:
+
+    * if `conn_spec` is "one_to_one", skip uniqueness check for better speed
+      and use data_connect
+    * if both `pre` and `post` are NodeCollections or can be converted to
+      NodeCollections (i.e. contain unique IDs), then proceed to "normal"
+      connect (potentially after conversion to NodeCollection)
+    * if both `pre` and `post` are arrays and contain non-unique items, then
+      we proceed to "data_connect"
+    * if at least one of them has non-unique items and they have different
+      sizes, then raise an error.
+    '''
+    data_connect = False
+
+    pre_is_nc, post_is_nc = True, True
+
+    if not isinstance(pre, NodeCollection):
+        # skip uniqueness check for data_connect compatible `conn_spec`
+        if conn_spec != 'one_to_one' and len(set(pre)) == len(pre):
```

Comment:
What if `conn_spec` is a dictionary? Isn't it unexpected that calling 
```python
nest.Connect(n, n, conn_spec='one_to_one')
```
behaves in one way, while calling
```python
nest.Connect(n, n, conn_spec={'rule': 'one_to_one'})
```
behaves differently?